internal/doc/modular_adoc_docs.go

```diff
@@ -275,35 +277,63 @@ func GenAsciidocCustom(cmd *cobra.Command, w io.Writer, linkHandler func(string)
 // If you have `cmd` with two subcmds, `sub` and `sub-third`,
 // and `sub` has a subcommand called `third`, it is undefined which
 // help output will be in the file `cmd-sub-third.1`.
-func GenAsciidocTree(cmd *cobra.Command, dir string) error {
-	identity := func(s string) string { return s }
-	emptyStr := func(s string) string { return "" }
-	return GenAsciidocTreeCustom(cmd, dir, emptyStr, identity)
+func GenAsciidocTree(cmd *cobra.Command, options GeneratorOptions) error {
+	return GenAsciidocTreeCustom(cmd, options)
+}
+
+// GeneratorOptions options for the generator
+type GeneratorOptions struct {
```

Comment:
Options file that let us control various aspects - we would typically provide flag in the cmd to support them from command line and enable generating content that suit all needs (even downstream)